bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/BiweeklyPresentableCalendar.java

```diff
@@ -167,6 +176,113 @@ public boolean isEventPresent(Instant instant) {
         return (this.getCurrentComponentWPeriod(instant) != null);
     }
 
+    @Override
+    public List<Event> getFilteredEventsBetween(Instant begin, Instant end, @Nullable EventTextFilter filter,
+            int maximumCount) {
+        List<VEventWPeriod> candidates = this.getVEventWPeriodsBetween(begin, end);
+        final List<Event> results = new ArrayList<>(candidates.size());
+
+        if (filter != null) {
+            Pattern filterPattern;
+            if (filter.type == Type.TEXT) {
+                filterPattern = Pattern.compile(".*" + Pattern.quote(filter.value) + ".*",
+                        Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
+            } else {
+                filterPattern = Pattern.compile(filter.value);
+            }
+
+            Class<? extends TextProperty> propertyClass;
+            switch (filter.field) {
+                case SUMMARY:
+                    propertyClass = Summary.class;
+                    break;
+                case COMMENT:
+                    propertyClass = Comment.class;
+                    break;
+                case CONTACT:
+                    propertyClass = Contact.class;
+                    break;
+                case DESCRIPTION:
+                    propertyClass = Description.class;
+                    break;
+                case LOCATION:
+                    propertyClass = Location.class;
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown Property to filter for.");
+            }
+
+            List<VEventWPeriod> filteredCandidates = candidates.stream().filter(current -> {
+                List<? extends TextProperty> properties = current.vEvent.getProperties(propertyClass);
+                for (TextProperty prop : properties) {
+                    if (filterPattern.matcher(prop.getValue()).matches()) {
+                        return true;
+                    }
+                }
+                return false;
+            }).collect(Collectors.toList());
+            candidates = filteredCandidates;
+        }
+
+        for (VEventWPeriod eventWPeriod : candidates) {
+            results.add(eventWPeriod.toEvent());
+        }
+
+        Collections.sort(results, (Event o1, Event o2) -> {
+            if (o1.start.equals(o2.start)) {
+                return 0;
+            } else if (o1.start.isBefore(o2.start)) {
+                return -1;
+            } else {
+                return 1;
+            }
+        });
+
```

Comment:
You could let `Event` implement `Comparable` and use `Instant.compareTo()`, instead of implementing your own compare method.

Reply:
You are right in case the start of the event will stay the data events will be sorted after. I'm not sure, whether this case is enough for future.

Reply:
Sorry, I don't understand your comment.

Reply:
I thought here into future, as maybe the sorting could also be different. But implementing things partly is bad design, so i'm making `Event` `Comparable`.